packages/server/test/security/api.test.ts

```diff
@@ -0,0 +1,1005 @@
+            webhooks: expect.anything()
+          })
+          expect(err.response?.status).toEqual(400)
+        }
+      )
+    })
+
+    test('Should not allow clientId other than valid UUID', async () => {
+      const ids = Array.from({ length: 10 }, () => froth(UUID_LENGTH))
+
+      for (const id of ids) {
+        await shouldFail(
+          async () => sync({ id }),
+          (err) => {
+            expect(err.response?.data).not.toEqual({
+              id: expect.anything(),
+              token: expect.anything(),
+              webhooks: expect.anything()
+            })
+            expect(err.response?.status).toEqual(400)
+          }
+        )
+      }
+    })
+
+    test('Should return a new set of credentials if clientId is unknown', async () => {
+      const id = uuid()
+
+      const res = await sync({ id })
+      expect(res.id).not.toEqual(id)
+    })
+
+    test('Should not be able to sync an empty channel config', async () => {
+      await shouldFail(
+        async () => sync({ channels: { teams: null } }),
+        (err) => {
+          expect(err.response?.data).not.toEqual({
+            id: expect.anything(),
+            token: expect.anything(),
+            webhooks: expect.anything()
+          })
+          expect(err.response?.status).toEqual(400)
+        }
+      )
+    })
+
+    test('Should not be able to make sync request with an heavy payload', async () => {
+      const str = 'a'.repeat(MAX_PAYLOAD_SIZE * 2)
+
+      await shouldFail(
+        async () => sync({ str } as any, { maxContentLength: Infinity, maxBodyLength: Infinity }),
+        (err) => {
+          expect(err.response?.data).toContain('PayloadTooLargeError: request entity too large')
+          expect(err.response?.status).toEqual(413)
+        }
+      )
+    })
+  })
+
+  describe('User', () => {
+    const user = async (clientId?: string, clientToken?: string, config?: AxiosRequestConfig) => {
+      const res = await http(clientId, clientToken).post<User>('/api/users', null, config)
+
+      expect(res.data).toEqual({ id: expect.anything(), clientId })
+      expect(res.status).toEqual(201)
+
+      return res.data
+    }
+
+    const getUser = async (userId: string, clientId?: string, clientToken?: string, config?: AxiosRequestConfig) => {
+      const res = await http(clientId, clientToken).get<User>(`/api/users/${userId}`, config)
+
+      expect(res.data).toEqual({ id: userId, clientId })
+      expect(res.status).toEqual(200)
+
+      return res.data
+    }
+
+    describe('Create', () => {
+      test('Should be able to create a user with valid credentials', async () => {
+        const res = await user(clients.first.clientId, clients.first.clientToken)
+
+        clients.first.userId = res.id
+
+        {
+          const res = await user(clients.second.clientId, clients.second.clientToken)
+
+          expect(res.id).not.toEqual(clients.first.userId)
+
+          clients.second.userId = res.id
+        }
```

Comment:
Why is there a random bracket here?

Reply:
To create another block of code. That way I can redefine a res constant without having to name it res2 or something